src/impl_methods.rs

```diff
@@ -1576,11 +1560,10 @@ where
     ///                                 [3, 4]]).into_dyn();
     /// ```
     pub fn into_dyn(self) -> ArrayBase<S, IxDyn> {
-        ArrayBase {
-            data: self.data,
-            ptr: self.ptr,
-            dim: self.dim.into_dyn(),
-            strides: self.strides.into_dyn(),
+        // safe because new dims equivalent
+        unsafe {
+            ArrayBase::from_data_ptr(self.data, self.ptr)
```

Comment:
```suggestion
            self
```

Reply:
Here it's a partial move - `into_dyn()` moves the D values.

Reply:
Oh, right. Sorry about that.